conan/internal/graph/graph_binaries.py

```diff
@@ -199,6 +201,24 @@ def _find_build_compatible_binary(self, node, compatibles):
         node.binary = original_binary
         node._package_id = original_package_id
 
+    def _filter_compat_remotes(self, node, remotes):
```

Comment:
I am not sure that explicit filtering the remotes is the way to go, I was thinking about the ``Remote._caching`` structure making it more powerful, but it also sounds challenging.